### PR TITLE
feat(backend): implement fridge controller

### PR DESCRIPTION
**Pull request by JJTomines**
- add fridge controller to get ingredients from database
- modify service to use ingredients from database
- modify component to use ingredients from database
- remove mock-ingredients used for testing

gh-11

### CODE REVIEW

File: client/src/services/fridge/fridge.service.ts [hunk 1]
```diff
@@ -1,43 +1,37 @@
...
 })
 export class FridgeService {
-  constructor() {}
 
-  public getAvailableIngredients(): string[] {
-    //TODO: Replace when Backend API available
-    return [...INGREDIENTS].map((ingredient) => ingredient.name).sort();
+  public apiIngredientsList: Ingredient[] = [];
+
+  constructor(private http: HttpClient, @Inject('BASE_URL') private baseUrl: string) {
```
Comment (StevieShibly8) [inline client/src/services/fridge/fridge.service.ts:17]: We have http-client.service set up that handles all the httpClient calls. You should be able to replace this with the service instead. Check recipe-service for reference.

File: client/src/services/fridge/fridge.service.ts [hunk 2]
```diff
@@ -1,43 +1,37 @@
...
   }
 
-  public getLocallyStoredIngredients(): Set<string> {
-    const localIngredients = localStorage.getItem('userIngredients');
-    if (localIngredients != null) {
+  getIngredientsData(): Observable<Ingredient[]> 
+  {
+    return this.http.get<Ingredient[]>(this.baseUrl + 'api/fridge');
```
Comment (StevieShibly8) [inline client/src/services/fridge/fridge.service.ts:22]: Once you replace httpClient with httpClientService, replace this line too to call the httpClientService's get method instead and only pass in for the url "fridge". httpClientService deals with the 'api/' route separately

File: client/src/services/fridge/fridge.service.ts (hunk 1 above)
Comment (JJTomines) [reply client/src/services/fridge/fridge.service.ts]: Updated to use http-client service!

File: server/Controllers/FridgeController.cs [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
+            _context = context;
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Test()
+        {
+            var ingredients = await _context.Ingredient.ToListAsync();
+            if (ingredients != null)
+            {
+                return Ok(ingredients);
```
Comment (zaidi-01) [inline server/Controllers/FridgeController.cs:27]: Return this right away. `ToListAsync` returns an empty list if no ingredients are found and that should be the desired behavior.

File: client/src/app/fridge/fridge.component.ts [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
   templateUrl: './fridge.component.html',
   styleUrls:['./fridge.component.scss']
 })
 export class FridgeComponent implements OnInit{
-  constructor(private fridgeService: FridgeService) {}
+  constructor(private fridgeService: FridgeService) {
+    this.getIngredients();
+  }
 
   addedIngredients: Set<string> = this.fridgeService.getLocallyStoredIngredients();
```
Comment (zaidi-01) [inline client/src/app/fridge/fridge.component.ts:14]: Variables should appear before the constructor. The order goes variables > constructor > methods, and public always before private.

**Review by zaidi-01 — APPROVED** [review]
Good work!

**Review by StevieShibly8 — APPROVED** [review]

**Pull request merged**